src/Model/GiftCardConfiguration.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Model;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\ImageInterface;
+use Sylius\Component\Resource\Model\TimestampableTrait;
+use Sylius\Component\Resource\Model\ToggleableTrait;
+
+class GiftCardConfiguration implements GiftCardConfigurationInterface
+{
+    use TimestampableTrait;
+    use ToggleableTrait;
+
+    /** @var int|null */
+    protected $id;
+
+    /** @var string|null */
+    protected $code;
+
+    /** @var ImageInterface[]|Collection */
+    protected $images;
+
+    /** @var GiftCardChannelConfigurationInterface[]|Collection */
+    protected $channelConfigurations;
+
+    /** @var bool|null */
+    protected $default = false;
+
+    public function __construct()
+    {
+        $this->images = new ArrayCollection();
+        $this->channelConfigurations = new ArrayCollection();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getCode(): ?string
```

Comment:
Maybe implement the `CodeAwareInterface` and I guess there is a trait also

Reply:
The interface is implemented (in GiftCardConfigurationInterface) but there is no such trait...

Reply:
Btw, missed useful traits like this should be added to SyliusCorePlugin.
But from other side, they can be used only at apps. Having SyliusCorePlugin as dependency at other plugins is not good.